### PR TITLE
type_matcher.rs: add support for: l b c p s

### PR DESCRIPTION
**Pull request by ilius**
Now all file types except for D (door on Solaris) are supported.

This is not blocked by the support of -H -L -P flags.
Because the default behavior of GNU find is "Never follow symbolic links".

```
-P     Never follow symbolic links.  This is the default behaviour.  When find examines or prints information
              about  files, and the file is a symbolic link, the information used shall be taken from the properties
              of the symbolic link itself.
```

### CODE REVIEW

**Comment by sylvestre** [comment]
Could you also please add tests? thanks

**Comment by ilius** [comment]
I'm still trying to test sym links on Windows.

**Comment by ilius** [comment]
Okay I added tests.

File: src/find/mod.rs [hunk 1]
```diff
@@ -104,6 +104,7 @@ fn parse_args(args: &[&str]) -> Result<ParsedInfo, Box<dyn Error>> {
 }
 
 #[allow(clippy::borrowed_box)] // FIXME?
+#[allow(clippy::unnecessary_wraps)] // FIXME?
```
Comment (sylvestre) [inline src/find/mod.rs:107]: Why did you add this?

Comment (ilius) [reply src/find/mod.rs:107]: The ci failed with `clippy::unnecessary-wraps` error

File: src/find/matchers/type_matcher.rs [hunk 2]
```diff
@@ -38,6 +45,15 @@ impl TypeMatcher {
         })
     }
 
+    pub fn is_symlink(file_type: &FileType) -> bool {
```
[the remark this answers is not in the document]
Comment (ilius) [reply src/find/matchers/type_matcher.rs:63]: To check -H -L -P flags later. You want me to directly use `FileType.is_symlink`?

[the remark this answers is not in the document]
Comment (sylvestre) [reply src/find/matchers/type_matcher.rs:63]: Yeah, No need for a function here

File: src/find/mod.rs (hunk 1 above)
Comment (sylvestre) [reply src/find/mod.rs:107]: Please fix the issue into a separate PR
A PR should focus on one thing :)
thanks

**Comment by ilius** [comment]
Updated.

**Comment by sylvestre** [comment]
Sorry for the latency :)
